frontend/packages/core/src/AppLayout/shortLinker.tsx

```diff
@@ -0,0 +1,199 @@
+ * - If successful,
+ * -     Will switch to a readable only input field with a clipboard button
+ *        to copy the short link with generated hash
+ *       (this is the only time it is ever displayed)
+ * - If not successful,
+ * -     Will display a toast error message for the user
+ */
+const ShortLinker = () => {
+  const { workflows } = useAppContext();
+  const { removeWorkflowSession, retrieveWorkflowSession } = useShortLinkContext();
+  const [open, setOpen] = React.useState(false);
+  const [shortLink, setShortLink] = React.useState<string | null>(null);
+  const [validWorkflow, setValidWorkflow] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<ClutchError | null>(null);
+  const anchorRef = React.useRef(null);
+  const location = useLocation();
+
+  /**
+   * Will fetch the the workflow based on the current route
+   * Then checks to see if the shortLink property of the workflow has been set.
+   *  - If it has, we will set the validWorkflow state which will allow the component to render
+   */
+  const checkValidWorkflow = () => {
+    const workflow = workflowByRoute(workflows, location.pathname);
+
+    setValidWorkflow(workflow?.shortLink ?? false);
+  };
+
+  // will trigger on a location change, emptying out our temporary storage and rechecking the workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+
+    removeWorkflowSession();
+  }, [location]);
+
+  // Used for initial load to verify that once our workflows have loaded we are on a valid workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+  }, [workflows]);
+
+  const handleToggle = () => {
+    setOpen(!open);
+    setShortLink(null);
+  };
+
+  const handleClose = event => {
+    if (anchorRef.current && anchorRef.current.contains(event.target)) {
+      return;
+    }
+    setOpen(false);
+  };
+
+  // Will rotate our object into an array of type IShareableState to send in the API request
+  const rotateStore = (sessionStore: HydratedData): IClutch.shortlink.v1.IShareableState[] =>
+    Object.keys(sessionStore).map(key => ({ key, state: sessionStore[key] }));
+
+  const generateShortLink = () => {
+    const requestData: IClutch.shortlink.v1.ICreateRequest = {
+      path: `${location.pathname}${location.search}`,
+      state: rotateStore(retrieveWorkflowSession()),
+    };
+
+    client
+      .post("/v1/shortlink/create", requestData)
+      .then(response => {
+        const { hash } = response.data as IClutch.shortlink.v1.ICreateResponse;
+        setShortLink(generateShortLinkRoute(window.location.origin, hash));
+      })
+      .catch((err: ClutchError) => setError(err));
+  };
+
+  if (!validWorkflow) {
+    return null;
+  }
+
+  return (
+    <>
+      {error && (
+        <Toast severity="error" onClose={() => setError(null)}>
+          Unable to generate short link, please try again later
```

Comment:
Should we use the Error component here to get more robust error information

Reply:
Sure, I'll give it a shot

Reply:
Modified to use the error component. Have to force it below the header but it works pretty well. Modified the component to also allow an `onClose` parameter so a user can get rid of it if they want to.

![Screen Shot 2022-04-27 at 1 38 25 PM](https://user-images.githubusercontent.com/8338893/165626970-c14a2e8c-f40c-4b19-8737-01c21ad778b6.png)

Reply:
Modified it to using a toast again and just deliberately putting the specific messaging in. Reverted the changes to the Error component.

![Screen Shot 2022-04-27 at 5 59 51 PM](https://user-images.githubusercontent.com/8338893/165805789-11b51c8e-91fe-40b3-8593-d13fec16e7d1.png)